test/nostrum/struct/user_test.exs

```diff
@@ -0,0 +1,15 @@
+defmodule Nostrum.Struct.UserTest do
+  use ExUnit.Case, async: true
+
+  alias Nostrum.Struct.User
+
+  doctest User
+
+  describe "String.Chars" do
```

Comment:
I think it's a little redundant to have a doctest and an actual test. What do you think? This is the case throughout this pr. Maybe it doesn't matter?

Reply:
They aren't redundant, but I do wish I could have made them doctests. They test functionality for the `String.Chars` protocol, but the protocols aren't directly doctestable IIRC. If there is a way to make them doctestable, I would be on board.